pkg/v1/cli/discovery/kubernetes.go

```diff
@@ -56,17 +57,28 @@ func (k *KubernetesDiscovery) Name() string {
 	return k.name
 }
 
-// Manifest returns the manifest for a local repository.
+// Manifest returns the manifest for a kubernetes repository.
 func (k *KubernetesDiscovery) Manifest() ([]plugin.Discovered, error) {
-	plugins := make([]plugin.Discovered, 0)
-
 	// Create cluster client
 	clusterClientOptions := clusterclient.Options{GetClientInterval: 2 * time.Second, GetClientTimeout: 5 * time.Second}
 	clusterClient, err := clusterclient.NewClient(k.kubeconfigPath, k.kubecontext, clusterClientOptions)
 	if err != nil {
 		return nil, err
 	}
 
+	return k.GetDiscoveredPlugins(clusterClient)
+}
+
+// GetDiscoveredPlugins returns the list of discovered plugin from a kubernetes cluster
+func (k *KubernetesDiscovery) GetDiscoveredPlugins(clusterClient clusterclient.Client) ([]plugin.Discovered, error) {
+	plugins := make([]plugin.Discovered, 0)
+
+	exists, err := clusterClient.VerifyCLIPluginCRD()
+	if !exists || err != nil {
```

Comment:
**question**: Looks like we're swallowing `err`, is this intentional?

Reply:
Thanks for the catch. Not it's not intentional. Updated